### ISSUE TITLE
What's the next estimated ASV release date?

### ISSUE THREAD

**Issue by anmyachev** [airspeed-velocity/asv#967]

**Comment by mlondschien** [comment]
We have published the current master as a developement release on conda-forge: https://github.com/conda-forge/asv-feedstock/pull/23. You should be able to install it with `conda install -c conda-forge/label/asv_dev asv`.

**Comment by QuLogic** [comment]
That doesn't really answer the question though...

**Comment by anmyachev** [comment]
@pv maybe you have a bunch of tasks that need to be done for the new release?

**Comment by mattip** [comment]
Are there any blockers to making a release now? Should we do a RC or just release the current HEAD to PyPI? Would it be v0.4.3? (the last release in May 2020 was v0.4.2). Browsing through the PRs merged since then it seems like they were bug fixes and there are no real command-line or API changes.

**Comment by mlondschien** [comment]
I believe to remember that the changes included a feature we were interested in using, which is why we we published the Dev release. I can't remember the details (I might figure this out), but would argue that a minor (not bugfix) release would be cleanest.

**Comment by mattip** [comment]
so v0.5.0?

**Comment by jbrockmendel** [comment]
Looks like the CI passed on #981, but the CI looks pretty bare-bones.  How confident should we be that we're in a release-able state?

**Comment by datapythonista** [comment]
I'm +1 on releasing with what we've got now. Short term plans include things like dropping py2 support, improving styles, splitting the UI and the core... I think it makes sense to release before all that.

@jbrockmendel What is the main concern here, that the CI is just running with a single element in the matrix, or that the test suite is insufficient? I think we can run the pandas benchmarks with asv master before releasing. And soon (not sure if before or after the release), I'd like to move the CI to GitHub actions (#985), have a proper matrix, linting, run the docs build... Do you have any other idea in mind?

**Comment by jbrockmendel** [comment]
> What is the main concern here, that the CI is just running with a single element in the matrix, or that the test suite is insufficient? 

That the test suite may be insufficient.  I haven't looked at it so have no idea what state it is in.

> I think we can run the pandas benchmarks with asv master before releasing.

This sounds like a reasonable plan.

**Comment by datapythonista** [comment]
@mattip @jbrockmendel does any of you want to release asv? I can take care of it if you prefer, but I need access to pypi (username datapythonista), and I guess to conda-forge, which I'm not sure if you've already got access.

**Comment by jbrockmendel** [comment]
I just invited you on pypi, no idea about conda-forge.

**Comment by astrojuanlu** [comment]
Current maintainers of asv-feedstock: https://github.com/conda-forge/asv-feedstock/blob/a534fd8cf431f069ec94e37dd51451f1e8b8cd56/recipe/meta.yaml#L60-L64

**Comment by datapythonista** [comment]
Thanks @jbrockmendel and @astrojuanlu 

If there are no objections, I'll release asv v0.5.0 early next week with the current master.

**Comment by datapythonista** [comment]
Starting the release of `v0.5.0`. Will post updates here

**Comment by datapythonista** [comment]
Pip package ready: https://pypi.org/project/asv/0.5/

The release is finally `0.5` (not `0.5.0`) to be consistent with previous releases.

Going to update conda-forge, not sure if today or maybe tomorrow.

**Comment by mlondschien** [comment]
The conda-forge update will be done automatically. You can open a PR adding yourself as a maintainer if you'd like.

**Comment by datapythonista** [comment]
Nice, I openened the PR for the conda-forge release manually already, and I added new project maintainers: https://github.com/conda-forge/asv-feedstock/pull/27

There seems to be an issue about `requirements-dev.txt` not being found (it's opened in `setup.py`). I need to have a look and see exactly what's the problem.

For pypi, I just upladed the source dist. I saw that for previous releases there were no binary wheels, so I guess that's enough, but just mentioning anyway for visibility.

Besides conda-forge, the pending thing is updating readthedocs. Not sure if this will happen automatically, or if something needs to be done. I guess I'll need permissions if doing something manual in readthedocs is required.

**Comment by xhochy** [comment]
> There seems to be an issue about requirements-dev.txt not being found (it's opened in setup.py).

This probably needs to be added to
MANIFEST.in to be part of the sdist.

**Comment by datapythonista** [comment]
> This probably needs to be added to MANIFEST.in to be part of the sdist.

Amazing, thanks a lot @xhochy very helpful. I opened #1013, will merge when CI is green. I guess it's ok to replace the git tag and the pypi package for 0.5 with this change? Or any reason to release 0.5.1 instead?

**Comment by xhochy** [comment]
I don't think you can replace sdists on PyPI, so this needs to be 0.5.1

**Comment by s-weigand** [comment]
You can do a [post release](https://www.python.org/dev/peps/pep-0440/#post-releases) `0.5.0.post0`
>Some projects use post-releases to address minor errors in a final release that do not affect the distributed software (for example, correcting an error in the release notes).

**Comment by datapythonista** [comment]
I completed the release of `asv 0.5.1`. Packages are available in the pypi, conda-forge and as a github release. I also updated the stable branch to `0.5.1` (thanks for the heads up @mattip).

> You can do a [post release](https://www.python.org/dev/peps/pep-0440/#post-releases) `0.5.post0`

Thanks for the info on this @s-weigand, very useful to know. I still preferred to use `0.5.1` in this case, as I thought it'd make things easier and no big difference, but will be useful in the future.

> If you're making a new (0.5.1) release, can you please also include #1002?

Sorry this didn't get in @QuLogic. For 0.5.1 it wasn't a good idea to introduce anything else than what fixed the problems in 0.5, but we'll be releasing a new version quite soon this time.
